sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/TestProxyRecordPolicy.java

```diff
@@ -78,11 +95,50 @@ private String serializeVariables(Queue<String> variables) {
         }
     }
 
+    @Override
+    public HttpResponse processSync(HttpPipelineCallContext context, HttpPipelineNextSyncPolicy next) {
+        TestProxyUtils.changeHeaders(context.getHttpRequest(), xRecordingId, "record");
+        return next.processSync();
+    }
+
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
         HttpRequest request = context.getHttpRequest();
         TestProxyUtils.changeHeaders(request, xRecordingId, "record");
         return next.process();
     }
+
+    private void addProxySanitization() {
+        this.sanitizers.forEach(testProxySanitizer  -> {
+            String requestBody;
+            String sanitizerType;
+            switch (testProxySanitizer.getType()) {
+                case URL:
+                    requestBody = createUrlRegexRequestBody(testProxySanitizer.getRegex(), testProxySanitizer.getRedactedValue());
+                    sanitizerType = TestProxySanitizerType.URL.name;
+                    break;
+                case BODY:
+                    requestBody = createBodyRegexRequestBody(testProxySanitizer.getRegex(), testProxySanitizer.getRedactedValue());
+                    sanitizerType = TestProxySanitizerType.BODY.name;
+                    break;
+                case HEADER:
+                    requestBody = createHeaderRegexRequestBody(testProxySanitizer.getRegex(), testProxySanitizer.getRedactedValue());
+                    sanitizerType = TestProxySanitizerType.HEADER.name;
+                    break;
+                default:
+                    throw new RuntimeException("Sanitizer type not supported");
+            }
+            addRegexSanitizer(requestBody, sanitizerType);
+        });
+    }
+
+    private void addRegexSanitizer(String requestBody, String sanitizerType) {
+        HttpRequest request
+            = new HttpRequest(HttpMethod.POST, String.format("%s/Admin/AddSanitizer", TestProxyUtils.getProxyUrl()))
+            .setBody(requestBody);
+        request.setHeader("x-abstraction-identifier", sanitizerType);
+        request.setHeader("x-recording-id", xRecordingId);
+        client.sendSync(request, Context.NONE);
+    }
```

Comment:
Could we factor this out into its own type and give it the client to avoid the duplication?

Reply:
moved to common utils.